eng/pipelines/release-cli.yml

```diff
@@ -768,6 +770,38 @@ stages:
           - bash: ls
             workingDirectory: hosting
 
+          - bash: |
+              unzip ./azd-darwin-amd64.zip -d ./tmp
+
+              # Ad-hoc signing with "-" identity
+              codesign -s - tmp/azd-darwin-amd64
+
+              zip azd-darwin-amd64.zip -j tmp/azd-darwin-amd64
+            displayName: Self-sign (Darwin)
+            condition: and(succeeded(), contains(variables['Agent.OS'], 'Darwin'))
+            workingDirectory: hosting
+
+          - pwsh: |
+              $ErrorActionPreference = 'Stop'
+
+              # Generate self-sign cert
+              $cert = New-SelfSignedCertificate -CertStoreLocation Cert:\LocalMachine\My -Type CodeSigningCert -Subject "azd installer tests code signing"
+              
+              # Add as temporary trusted root CA
+              try {
+                Export-Certificate -Cert $cert -FilePath code_signing.crt
+                Import-Certificate -FilePath .\code_signing.crt -Cert Cert:\LocalMachine\Root
+              }
+              finally {
+                Remove-Item -Force .\code_signing.crt -ErrorAction SilentlyContinue
+              }
+
+              # Sign the windows binary
+              Set-AuthenticodeSignature .\azd-windows-amd64.msi -Certificate $cert
+            displayName: Self-sign (Windows)
+            condition: and(succeeded(), contains(variables['Agent.OS'], 'Windows'))
+            workingDirectory: hosting
+
```

Comment:
Both the `bash` and `pwsh` scripts could be written into a single `.ps1` file that checks `$IsWindows` and `$IsMacOS` (and noops on `$IsLinux`).

It could then be called with something like: 

```yaml
- task: PowerShell@2 
  inputs: 
    pwsh: true
    targetType: filePath 
    filePath: eng/scripts/Set-SelfSignedCerts.ps1
    arguments: -ArtifactDirectory "hosting"
```


I differentiated `bash` and `pwsh` with `http-server` below because we couldn't find a reliable way to use pwsh to launch `http-server` (or any background process) on a non-Windows OS.